buildcontext/gitlookup.go

```diff
@@ -25,14 +28,14 @@ import (
 )
 
 type gitMatcher struct {
-	name     string
-	re       *regexp.Regexp
-	sub      string
-	user     string
-	suffix   string
-	protocol gitProtocol
-	password string
-	keyScan  string
+	name              string
+	re                *regexp.Regexp
+	sub               string
+	user              string
+	suffix            string
+	protocol          gitProtocol
+	password          string
+	allowEmptyKeyScan bool
```

Comment:
nit: insecureAllowUnknownHosts

Reply:
I'm hesitant to extend these changes here, as it makes the rest of the code rather verbose to read, for example:

```
		if len(keyScans) == 0 && !m.insecureAllowEmptyKeyScan {
			return "", nil, errors.Errorf("no known_hosts entries exist for %s", host)
		}
```

which I then read as "if not insecure" which reads like a double-negative.

setting `allowEmptyKeyScan` is essentially ssh's `StrictHostKeyChecking` setting. Maybe it's better to just use the later as the variable name?

Then this becomes:

```
		if len(keyScans) == 0 && m.StrictHostKeyChecking {
			return "", nil, errors.Errorf("no known_hosts entries exist for %s", host)
		}
```

Reply:
I made the above change, I think it's easier to explain to our users what is actually happening (better to re-use existing terminology).

Reply:
Nice, I like StrictHostKeyChecking better too :+1: